### ISSUE TITLE
Provide maskrcnn-benchmark trainer as example

### ISSUE THREAD

**Issue by vfdev-5** [pytorch/ignite#330]
I think this will be a great use-case for ignite as a tool. 
Especially, we can simplify some stuff and provide more features. For example:

- Replace [WarmupMultiStepLR](https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/maskrcnn_benchmark/solver/lr_scheduler.py#L7) by our ConcatScheduler = some WarmupLR + MultiStepLR
- [Trainer](https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/maskrcnn_benchmark/engine/trainer.py) -> our engine with features
- Logging -> ProgressBar, TB logger or Visdom logger
- Checkpointing
- etc

What do you guys think about ?

**Comment by miguelvr** [comment]
Love the idea. I'm personally using the tool already so I am happy to help

**Comment by miguelvr** [comment]
Sounds good to me! You mean a branch in their repo, right?

**Comment by vfdev-5** [comment]
@miguelvr I meant to create a branch here. At least at the first time before we prove that it becomes useful and then we can propose something to @fmassa

**Comment by miguelvr** [comment]
So is the idea to create a clone folder and change it there?

**Comment by vfdev-5** [comment]
@miguelvr I think we could try to install maskrcnn-benchmark as library following their install suggestion and rewrite trainer, reusing all other parts...

**Comment by miguelvr** [comment]
sounds good! let's do it

**Comment by vfdev-5** [comment]
Here is the [branch](https://github.com/pytorch/ignite/tree/maskrcnn-example)

**Comment by vfdev-5** [comment]
On which dataset would you like to work? For a smoke test, maybe we would better to use VOC12 ?

**Comment by miguelvr** [comment]
I'm not a big expert in Computer Vision, just started experimenting recently... I'm more of an NLP/Speech guy. Anything easy to iterate on would be good!

**Comment by labor00** [comment]
The maskrcnn example would be a excellent  opportunity to use the ignite library in a distributed setting.
The metrics function would necessity some small changes to perform reduction of parameters and lr schedulers like reduceOnPlateau that depends of some quantity that probably also  need to be reduced.

**Comment by anmolsjoshi** [comment]
@vfdev-5 any help wanted? would love to help out with this!

**Comment by vfdev-5** [comment]
@anmolsjoshi thanks for the proposition ! I think a great improvement for this example will be the usage of Tensorboard or Visdom. I put some code [here](https://github.com/pytorch/ignite/compare/maskrcnn-example...vfdev-5:maskrcnn-example?expand=1) to start training. Next I need to adapt the inference part and then I can send a PR into mask-rcnn branch, so you can tryout your TB handler implementation. Let's discuss about TB handler integration once I've pushed the code.

**Comment by vfdev-5** [comment]
Close the issue as wont fix

**Comment by fran6co** [comment]
@vfdev-5 why was this marked as wont fix?

**Comment by vfdev-5** [comment]
@fran6co maskrcnn-benchmark, IMO, is an intermediate state to detectron2 and detection models in torchvision. I think it would be better to workout detectron2 code instead of maskrcnn-benchmark.

**Comment by fran6co** [comment]
ah, I thought this was a catch all issue for mask rcnn
